core/tee/tee_svc_cryp.c

```diff
@@ -1664,27 +1673,27 @@ static TEE_Result copy_in_attrs(struct user_ta_ctx *utc,
 	TEE_Result res = TEE_SUCCESS;
 	size_t size = 0;
 	uint32_t n = 0;
+	struct utee_attribute *usr_attrs_bbuf = NULL;
 
 	if (MUL_OVERFLOW(sizeof(struct utee_attribute), attr_count, &size))
 		return TEE_ERROR_OVERFLOW;
 
-	usr_attrs = memtag_strip_tag_const(usr_attrs);
+	usr_attrs_bbuf = bb_alloc(size);
+	if (!usr_attrs_bbuf)
+		return TEE_ERROR_OUT_OF_MEMORY;
 
-	res = vm_check_access_rights(&utc->uctx,
-				     TEE_MEMORY_ACCESS_READ |
-				     TEE_MEMORY_ACCESS_ANY_OWNER,
-				     (uaddr_t)usr_attrs, size);
-	if (res != TEE_SUCCESS)
+	res = copy_from_user(usr_attrs_bbuf, usr_attrs, size);
+	if (res)
```

Comment:
Prefer `goto out` instead of `return res` at line below, for consistency.

Reply:
oh yes, thanks! updated